src/server/game/Entities/Object/Object.h

```diff
@@ -487,6 +487,7 @@ class WorldObject : public Object, public WorldLocation
         bool IsWithinDistInMap(WorldObject const* obj, float dist2compare, bool is3D = true) const;
         bool IsWithinLOS(float x, float y, float z) const;
         bool IsWithinLOSInMap(WorldObject const* obj) const;
+        virtual Position GetHitSpherePointFor(Position const dest) const;
```

Comment:
Why do you use virtual polymorphisms here instead of checking for player in whatever function calls it?

IMO it would be cleaner to have the "proper" method here and then just choose not to use it for players in the calling method.